activerecord/lib/active_record/validations.rb

```diff
@@ -63,12 +63,17 @@ def valid?(context = nil)
 
   protected
 
+    def run_validations? # :nodoc:
+      @run_validations
+    end
+
     def raise_validation_error
       raise(RecordInvalid.new(self))
     end
 
     def perform_validations(options={}) # :nodoc:
-      options[:validate] == false || valid?(options[:context])
+      @run_validations = options[:validate] != false
```

Comment:
> Like @rafaelfranca I'm not sure whether we want to do this...

I'm reading the conversation again and again but I can't see anything about he don't want to fix this.
Passing `:validate` option through all the methods does not make sense because the stacktrace of the `autosave_associations` methods doesn't include only ActiveRecord methods. Maybe keeping `:validate` option in Transaction information makes more sense.